rakelib/form526.rake

```diff
@@ -226,4 +226,127 @@ namespace :form526 do
+      # gsub with a block explicitly sets backrefs correctly https://ruby-doc.org/core-2.6.6/String.html#method-i-gsub
+      json.gsub!(OliveBranch::Middleware::VA_KEY_REGEX) do
+        key = $1
+        "#{key.gsub('VA', 'Va')}:"
+      end
+      JSON.parse(json)
+      # rubocop:enable Style/PerlBackrefs
+    end
+
+    # make a hash that has both the original and corrupted versions of the disability name.
+    # eg {"myocardial infarction (mi)" => "myocardial infarction (MI)"}
+    def get_disability_array(form_data_hash)
+      new_conditions = form_data_hash['newDisabilities']&.collect { |d| d.dig('condition') } || []
+      rated_disabilities = form_data_hash['ratedDisabilities']&.collect { |rd| rd['name'] } || []
+      new_conditions + rated_disabilities
+    end
+
+    # downcase and remove everything but letters and numbers
+    def simplify_string(string)
+      string&.downcase&.gsub(/[^a-z0-9]/, '')
+    end
+
+    def get_dis_translation_hash(disability_array)
+      dis_translation_hash = {}
+      disability_array.each do |dis|
+        dis_translation_hash[simplify_string(dis)] = dis
+      end
+      dis_translation_hash
+    end
+
+    def fix_treatment_facilities_disability_name(form_data_hash, dis_translation_hash, disability_array)
+      transformed = false
+      # fix vaTreatmentFacilities -> treatedDisabilityNames
+      # this should never happen, just want to confirm
+      form_data_hash['vaTreatmentFacilities']&.each do |va_treatment_facilities|
+        new_treated_disability_names = {}
+        if va_treatment_facilities['treatedDisabilityNames']
+          va_treatment_facilities['treatedDisabilityNames'].each do |disability_name, value|
+            if disability_array.include? disability_name
+              new_treated_disability_names[disability_name] = value
+            else
+              transformed = true
+              original_disability_name = dis_translation_hash[simplify_string(disability_name)]&.downcase
+              new_treated_disability_names[original_disability_name] = value unless original_disability_name.nil?
+            end
+          end
+          va_treatment_facilities['treatedDisabilityNames'] = new_treated_disability_names
+        end
+      end
+      transformed
+    end
+
+    def fix_pow_disabilities(form_data_hash, dis_translation_hash, disability_array)
+      transformed = false
+      # just like treatedDisabilityNames fix the same checkbox data for POW disabilities
+      pow_disabilities = form_data_hash.dig('view:isPow', 'powDisabilities')
+      if pow_disabilities
+        new_pow_disability_names = {}
+        pow_disabilities.each do |disability_name, value|
+          if disability_array.include? disability_name
+            new_pow_disability_names[disability_name] = value
+          else
+            transformed = true
+            original_disability_name = dis_translation_hash[simplify_string(disability_name)]&.downcase
```

Comment:
same as https://github.com/department-of-veterans-affairs/vets-api/pull/5651/files#r556663572